src/main/java/io/kestra/plugin/aws/eventbridge/PutEvents.java

```diff
@@ -0,0 +1,157 @@
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResponse;
+
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send multiple custom events as maps to Amazon EventBridge so that they can be matched to rules",
+            code = {
+                "entries:",
+                "  - eventBusName: \"events\"",
+                "    source: \"Kestra\"",
+                "    detailType: \"my_object\"",
+                "    detail:",
+                "      message: \"hello from EventBridge and Kestra\"",
+            }
+        ),
+        @Example(
+            title = "Send multiple custom events as a JSON string to Amazon EventBridge so that they can be matched to rules.",
+            code = {
+                "entries:",
+                "  - eventBusName: \"events\"",
+                "    source: \"Kestra\"",
+                "    detailType: \"my_object\"",
+                "    detail: \"{\\\"message\\\": \\\"hello from EventBridge and Kestra\\\"}\"",
+                "    resources:",
+                "      - \"arn:aws:iam::123456789012:user/johndoe\"",
+            }
+        )
+    }
+)
+@Schema(
+    title = "Send multiple custom events to Amazon EventBridge so that they can be matched to rules."
+)
+public class PutEvents extends AbstractConnection implements RunnableTask<PutEvents.Output> {
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @Schema(
+        title = "List of event entries to send to EventBridge",
+        description = "A list of at least one EventBridge entry."
+    )
+    private List<Entry> entries;
+
+    @Override
+    public PutEvents.Output run(RunContext runContext) throws Exception {
+        final long start = System.nanoTime();
+
+        try (var eventBridgeClient = client(runContext)) {
+            List<PutEventsRequestEntry> requestEntries = new ArrayList<>(entries.size());
+
+            for (Entry entry : entries) {
+                requestEntries.add(entry.toRequestEntry(runContext));
+            }
+
+            PutEventsRequest eventsRequest = PutEventsRequest.builder()
+                .entries(requestEntries)
+                .build();
+
+            PutEventsResponse putEventsResponse = eventBridgeClient.putEvents(eventsRequest);
+
+            runContext.metric(Timer.of("duration", Duration.ofNanos(System.nanoTime() - start)));
```

Comment:
Duration are in second like 1.123456 by convention on metrics not in nano

Reply:
ok, I stole it from the lambda one I was not sure :sweat_smile:  :+1:

Reply:
I checked but ofNanos transform it in seconds it seems ?

`public static Duration ofNanos(long nanos) {
        long secs = nanos / 1000000000L;
        int nos = (int)(nanos % 1000000000L);
        if (nos < 0) {
            nos = (int)((long)nos + 1000000000L);
            --secs;
        }

        return create(secs, nos);
    }`